lib/ruby_lsp/requests/folding_ranges.rb

```diff
@@ -166,11 +166,6 @@ def on_begin(node)
         add_simple_range(node)
       end
 
-      sig { params(node: YARP::Node).void }
-      def on_node(node)
```

Comment:
In order to group requires for a folding range, we need to accumulate call nodes that match `require` or `require_relative`. We keep expanding the range until we find any other type of node, which is on this removed `on_node` method. Then we emit the range.

We might be missing a test for this. Is it working for this case?

```ruby
require "foo"
require "bar"

class Baz
  def something; end
end

require "qux"
require "zod"
```
In that example, you should be have 3 folding ranges:
1. The two requires for foo and bar
2. The class
3. The two other requires for qux and zod